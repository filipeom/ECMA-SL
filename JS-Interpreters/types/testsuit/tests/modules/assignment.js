a = 5     // retval = 5
a += 5    // retval = 10
a -= 5    // retval = 5
a *= 5    // retval = 25
a /= 5    // retval = 5
a %= 3    // retval = 2
a **= 3   // retval = 8
a         // retval = 8



// retval = 8