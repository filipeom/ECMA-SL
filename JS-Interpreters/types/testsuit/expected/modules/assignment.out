PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Program [stmts]:"
PROGRAM PRINT: ["$loc_3", "$loc_7", "$loc_11", "$loc_15", "$loc_19", "$loc_23", "$loc_27", "$loc_31"]
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_4" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_6", "left": "$loc_5", "operator": "=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "AssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_8" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_10", "left": "$loc_9", "operator": "+=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "PlusAssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 10.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_12" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_14", "left": "$loc_13", "operator": "-=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "MinusAssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_16" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_18", "left": "$loc_17", "operator": "*=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "TimesAssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 25.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_20" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_22", "left": "$loc_21", "operator": "/=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "DivisionAssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "5", "value": 5 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_24" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_26", "left": "$loc_25", "operator": "%=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ModulusAssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "3", "value": 3 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 3.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 2.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_28" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_30", "left": "$loc_29", "operator": "**=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "PowerAssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "3", "value": 3 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 3.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 8.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_32" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Identifier [name][value]:"
PROGRAM PRINT: "a"
PROGRAM PRINT: 8.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 8.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "End of the program [retval]:"
PROGRAM PRINT: 8.
